client/driver/docker.go

```diff
@@ -166,6 +167,32 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task) (do
 	d.logger.Printf("[DEBUG] driver.docker: using %d cpu shares for %s", hostConfig.CPUShares, task.Config["image"])
 	d.logger.Printf("[DEBUG] driver.docker: binding directories %#v for %s", hostConfig.Binds, task.Config["image"])
 
+	//  set privileged (fallback to false)
+	hostConfig.Privileged, _ = strconv.ParseBool(task.Config["privileged"])
```

Comment:
This is a better example: https://github.com/hashicorp/nomad/blob/master/client/driver/raw_exec.go#L50